### PR TITLE
Bug 2015115: [RFE] PCI passthrough - Add HW to create VM review

### PR DESCRIPTION
**Pull request by avivtur**
Adding Hardware devices (GPUs & Host devices) to simple VM create wizard review.

With devices:

simple-review-2

simple-review
No devices:

simple-review-empty-2

simple-review-empty

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

### CODE REVIEW

**Comment by avivtur** [comment]
@yaacov please review

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+type VMIPProps = {
+  data: string[];
+  title?: string;
+};
+
+const VMHW: React.FC<VMIPProps> = ({ data, title }) => {
+  const { t } = useTranslation();
+  if (!data) {
+    return <>{t('kubevirt-plugin~None')}</>;
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:15]:
```suggestion
    return t('kubevirt-plugin~None');
```

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { useTranslation } from 'react-i18next';
+
+import './create-vm-form.scss';
+
+type VMIPProps = {
+  data: string[];
+  title?: string;
+};
+
+const VMHW: React.FC<VMIPProps> = ({ data, title }) => {
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:12]:
```suggestion
const VirtualMachineHardware: React.FC<VirtualMachineHardwareProps> = ({ data, title = 'Devices' }) => {
```

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  const isMoreThenOneHW = data?.length > 1;
+  return (
+    <div>
+      <div>{data?.[0]}</div>
+      {isMoreThenOneHW && (
+        <Popover
+          headerContent={
+            <div>
+              {title || t('kubevirt-plugin~Devices ({{devices}})', { devices: data.length })}
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:26]:
```suggestion
              {title}
```

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { Button, ButtonVariant, Popover, PopoverPosition } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+
+import './create-vm-form.scss';
+
+type VMIPProps = {
+  data: string[];
+  title?: string;
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:9]:
```suggestion
  title?: React.ReactNode;
```

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx (hunk 1 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:15]: maybe even return `emptyStateText: React.RactNode`

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx (hunk 3 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:26]: you know `data.length` in the caller method

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+const VMHW: React.FC<VMIPProps> = ({ data, title }) => {
+  const { t } = useTranslation();
+  if (!data) {
+    return <>{t('kubevirt-plugin~None')}</>;
+  }
+
+  const isMoreThenOneHW = data?.length > 1;
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:18]: btw/ it's a nice pattern to have a list with (+ N objects ) button as a common component

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { Button, ButtonVariant, Popover, PopoverPosition } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+
+import './create-vm-form.scss';
+
+type VMIPProps = {
+  data: string[];
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:8]:
```suggestion
  devicesNames: string[];
```

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm/forms/VMHW.tsx:8]: unless you want to make it a common component and then it's just `names`

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/create-vm-form.tsx [hunk 7]
```diff
@@ -53,9 +53,9 @@ import {
   URLSource,
 } from '../../vm-templates/vm-template-source';
 import { BootSourceState } from './boot-source-form-reducer';
-import { FORM_ACTION_TYPE, FormAction, FormState } from './create-vm-form-reducer';
-
+import { FormAction, FormState, FORM_ACTION_TYPE } from './create-vm-form-reducer';
 import './create-vm-form.scss';
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/create-vm-form.tsx:57]: - scss imports are last
- if possible (e.g. linting pass) have an empty line to separate from regular imports

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/create-vm-form.tsx [hunk 8]
```diff
@@ -80,6 +80,13 @@ export const CreateVMForm: React.FC<CreateVMFormProps> = ({
 }) => {
   const { t } = useTranslation();
   const { name, nameValidation, namespace, startVM, template } = state;
+  const gpus = template?.objects?.[0]?.spec?.template?.spec?.domain?.devices?.gpus?.map(
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/create-vm-form.tsx:83]:
```suggestion
  const devices = template?.objects?.[0]?.spec?.template?.spec?.domain?.devices;
  const gpus = devices && devices?.gpus?.map(
```

File: frontend/packages/kubevirt-plugin/src/components/create-vm/forms/create-vm-form.tsx [hunk 9]
```diff
@@ -274,6 +280,24 @@ export const CreateVMForm: React.FC<CreateVMFormProps> = ({
...
+                  devicesNames={gpus}
+                  title={t('kubevirt-plugin~GPU Devices ({{devices}})', { devices: gpus?.length })}
+                />
+              </FormRow>
+            </SplitItem>
+            <SplitItem>
+              <FormRow fieldId="host-devices" title={t('kubevirt-plugin~Host devices')}>
+                <VirtualMachineHardware
+                  devicesNames={hostDevices}
+                  title={t('kubevirt-plugin~Host Devices ({{devices}})', {
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm/forms/create-vm-form.tsx:295]:
```suggestion
                  title={t('kubevirt-plugin~Host Devices ({{numberOfDevices}})', {
```

**Pull request merged**